### DISCUSSION TITLE
Add hotkey to increase or decrease file pane font using keyboard

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1513] [Suggestions]
Being that DC is generally keyboard based, would anyone else find it useful to be able to add DC commands / hotkeys to be able to increase/ decrease the font size used in the file pane?
I know DC uses font size increase/decrease and not an actual zoom, but this would allow users the ability to assign the typical zoom based keyboard shortcuts for this
Ctrl+minus / Ctrl+Shift + equal (+)
or
Ctrl+minus / Ctrl+equal (+)

Hopefully others will find this useful as well
Thank You

**Comment by lars18th** [comment]
This is a good idea. I suggest to link `<CTRL>+<->` and `<CTRL>+<+>` to the `<CTRL>+Mouse_Wheel` actions. I feel it could be easy inside the source code.

**Comment by advice2020** [comment]
I know this is not possible but I just wanted to mention it anyways.
Taking this idea one step further I was thinking it might be nice to also add an Option to be able to link File Type / MIME icon sizes to specific DC "Main Font" (file pane) font size value ranges.
I was thinking this would add to the mimicing of "zoom" (even though it is not zooming)
Not fully accurate but I was thinking ranges
FONT SIZE = ICON SIZE
0-16 = 16x16
17-20 = 24x24
21-26 = 36x36
27-Max = 48x48
I played around with this a little to figure out the font size ranges, setting the icon size and then restarting DC over and over each time.  Took me a little while to realize.... if I have to restart DC to get the icon sizes to change, then ummmm..... dammit! :)
Anyways, if there is ever a way to get icon sizes to be able to change without having to restart DC, this might be a nice visual effect. :)
Thank You

**Comment by lars18th** [comment]
Hi @advice2020 ,

Interesting idea. Perhaps with small effort this could be implemented.

**Comment by Skif-off** [comment]
Guys, how many times a day do you do this? :)) And why so often that this requires additional ways? :)

@advice2020 

> Anyways, if there is ever a way to get icon sizes to be able to change without having to restart DC, this might be a nice visual effect.

Icon size used as an example in the [DC help files](https://doublecmd.github.io/doc/en/configuration.html#ConfigDC).

**Reply by G-i-o** [reply]
> Guys, how many times a day do you do this? :)) And why so often that this requires additional ways? :)

Quite often actually as it happens frequently that I accidentally hit the CTRL and TOUCHPAD together which causes the zoom level to go astray.
Then, reverting back to the precise level, or even close, it's quite tricky and annoying... with my TOUCHPAD at least.

This feature or the ability to lock the zoom at least would be absolutely welcome by me.

**Comment by advice2020** [comment]
Skif! I finally had another user agreeing with a suggestion of mine and you scared them away :) (just kidding)
But you don't use zooming (file size increasing in DC) in applications?  This is fairly common in many applications: file managers, text editors, web browsers, terminals, etc. so it is not that rare though.
Also because DC is said to be more keyboard based, having a keyboard hotkey seems like it would fit in, and this feature does exist with a mouse action.
The icon size is not as important, that was just an idea to have option to link the two, but not needed, just the fact that DC allows this is great.  
Also I saw link you mentioned, didn't realize that icon size/other items could change with restart in this way, but in this way probably would not work for a zoom feature, but again icon zoom not as important.

Anyways, well I thought it might have been a good idea and so did one other user :)
Okay, well thank you for the link / pointing out some capabilities of those apply/edit buttons in Options.

**Comment by ags1234** [comment]
@advice2020, I also agree with what you proposed and like to see such an easy to use function.
Many programs use CTRL+plus and CTRL+minus for zoom in/out. Or CTRL+ mouse wheel. Web browsers, for example.

**Comment by Skif-off** [comment]
@advice2020

> Also because DC is said to be more keyboard based

This is not entirely true :) [For example](https://github.com/doublecmd/doublecmd/discussions/1693).

@ags1234

> Or CTRL+ mouse wheel.

Ctrl+mouse wheel uses in [many places](https://doublecmd.github.io/doc/en/faq.html#cs_size).

P.S.
<details>

> Web browsers, for example.

I guess I will never understand why people compare a file manager to a web browser instead of other file managers :))
</details>

**Comment by ags1234** [comment]
> I guess I will never understand why people compare a file manager to a web browser instead of other file managers :))

We are thinking at the same thing but from different angles.
I can explain why I think my comparison is justified.
Being the user and working with many programs, it is easier to find that shortcuts that I already know and use in some programs, also work in others and do the same things. It is a feeling of ease, uniformity, familiarity.

Also, a file manager helps the user manage and browse content. A web browser does exactly the same thing, but less managing, more browsing :)
Of course they are different kinds of programs, but at this level they are perfectly comparable.
You get what I mean :)

**Comment by advice2020** [comment]
Ahh this continues to be one of my most active posts :)

@ags1234 
Thank you for agreeing with my request.
Also well said.
It is based on the things that programs have in common, nice to be able to use the same hotkeys to perform these actions.  How "Tabs" function is another good example.

@Skif-off 
I knew you were going to throw that in my face Skif :)
That mouse update means a lot to me, and now you use it against me :)
No I know, but you guys in the past often mentioned when I bring up mouse based functionality that DC is more keyboard centric.  I always said, hybrid approach is best :)
Speaking of which I agree, I have mentioned in the past that I really like how in DC you can use Ctrl+MSW on many different text items, I wish more programs would do this, but I also like a keyboard option as well.  I just want both :)

Thanks again guys for the responses

**Comment by Skif-off** [comment]
The only thing that came to my mind: a long time ago I was very impressed by a table with all the hot keys and modifier keys in PhotoShop. And all of them did not cancel the mouse as a tool.
But this is just a thought out loud :))

**Comment by advice2020** [comment]
Still interested in this one.
Okay if the icon size does not change in size, not really needed and would then be different from how the Ctrl+MSW currently functions in DC where it just adjusts the font size.
But the keyboard hotkey to achieve this would be great.
If this is not decided to be added at all, which is probably the case, by any chance does anyone know of any tricks/Lua approaches that maybe could allow this to be achieved using keyboard hotkeys such as...
Ctrl+minus
Ctrl+plus (or better if equal sign because then no Shift modifier needed)

Thank You

**Reply by G-i-o** [reply]
I very welcome your suggestion @advice2020 and hope it gets implemented at some point. (see [here](https://github.com/doublecmd/doublecmd/discussions/1513#discussioncomment-13217127) why:).

This or at least the ability to lock the file panel zoom level.

**Comment by advice2020** [comment]
This appears to now be possible.
Thank you for adding these commands
